### PR TITLE
feat: add lazy load to article

### PR DESCRIPTION
**Pull request by craigbilner**
We implemented the `lazy-load` package on list pages to improve the UX for people who scroll on a poor network. This PR implements it on the article page before we refactor for templates.

n.b. there's an additional improvement we could make by not requesting related article images if they're not visible.

feat: add size to native lead asset and use both resolutions for web
feat: add `registerNode` and `observed` to article body images
feat: add `id` to related articles container to determine when they're visible
feat: minimise `SSR` output for better perf testing
feat: pass `AUTH_TOKEN` envar to TPA for unteased articles
fix: if you provide both high and low res sizes to image it didn't render the high res one
refactor: remove redundant article row for web and use `renderTrees` instead. This gives us better "indexes for keys"
refactor: split out `Article` prop-types (still need some love around xxxpress events)
refactor: cache `styled-components` in `slice` constructors to avoid dismounting when props change
refactor: move card fade in to consumer (only used on web lists atm)
refactor: remove keys/indexes from child components (only needed for top level in lists)
chore: add stories around images
chore: add `lowResSize` to `article-image`

### CODE REVIEW

File: packages/article-image/article-image.showcase.js [hunk 1]
```diff
@@ -16,12 +16,30 @@ export default {
...
-      component: () => (
-        <ArticleImage
-          captionOptions={primaryImage.captionOptions}
-          imageOptions={primaryImage.imageOptions}
-        />
-      )
+      // eslint-disable-next-line react/prop-types
+      component: ({ boolean }) => {
+        const withHighRes = boolean("As high resolution");
+        const withLowRes = boolean("As low resolution");
```
Comment (colincclark) [inline packages/article-image/article-image.showcase.js]: knobs are a bit confusing. Surely just one checkbox is needed for high res? default would be low res? if both are checked it is odd

File: packages/article-list/src/article-list-item.web.js [hunk 2]
```diff
@@ -110,7 +110,7 @@ const ArticleListItem = props => {
           lowResSize={lowResSize}
           showImage={showImage}
         >
-          {children}
+          <Animations.FadeIn>{children}</Animations.FadeIn>
```
Comment (colincclark) [inline packages/article-list/src/article-list-item.web.js]: doesn't `Card` do the animation already?

File: packages/article/__tests__/web/article-lazy.web.test.js [hunk 3]
```diff
@@ -0,0 +1,220 @@
...
+          onLinkPress={() => {}}
+          onRelatedArticlePress={() => {}}
+          onTopicPress={() => {}}
+          onTwitterLinkPress={() => {}}
+          onVideoPress={() => {}}
+        />
+      );
+
+      const [
+        ,
```
Comment (colincclark) [inline packages/article/__tests__/web/article-lazy.web.test.js]: not sure what is happening here

File: packages/card/__tests__/web/__snapshots__/card.web.test.js.snap [hunk 4]
```diff
@@ -499,12 +443,10 @@ exports[`15. card should re-render when loading state changes 1`] = `
...
-      <div>
-        <div />
-        <div />
-        <div />
-        <div />
-      </div>
+      <div />
+      <div />
+      <div />
+      <div />
```
Comment (colincclark) [inline packages/card/__tests__/web/__snapshots__/card.web.test.js.snap]: this seems weird

File: packages/article-list/src/article-list-item.web.js (hunk 2 above)
Comment (craigbilner) [reply packages/article-list/src/article-list-item.web.js]: :speaking_head: this is specifically the content of the card

File: packages/article/__tests__/web/article-lazy.web.test.js (hunk 3 above)
Comment (craigbilner) [reply packages/article/__tests__/web/article-lazy.web.test.js]: destructuring the second item :-)

File: packages/article-image/article-image.showcase.js (hunk 1 above)
Comment (craigbilner) [reply packages/article-image/article-image.showcase.js]: np :+1:

File: packages/card/__tests__/web/__snapshots__/card.web.test.js.snap (hunk 4 above)
Comment (craigbilner) [reply packages/card/__tests__/web/__snapshots__/card.web.test.js.snap]: looks like the genuine structure? I've made a note to modernise these tests though

**Pull request merged**